data/helpers.d/backend

```diff
@@ -183,3 +183,76 @@ ynh_remove_fpm_config () {
 	ynh_secure_remove "/etc/php5/fpm/conf.d/20-$app.ini" 2>&1
 	sudo systemctl reload php5-fpm
 }
+
+# Create a dedicated fail2ban config (jail and filter conf files)
+#
+# usage: ynh_add_fail2ban_config log_file filter [max_retry [ports]]
+# | arg: -l, --logpath= - Log file to be checked by fail2ban
+# | arg: -r, --failregex= - Failregex to be looked for by fail2ban
+# | arg: -m, --max_retry= - Maximum number of retries allowed before banning IP address - default: 3
+# | arg: -p, --ports= - Ports blocked for a banned IP address - default: http,https
+ynh_add_fail2ban_config () {
+  # Declare an array to define the options of this helper.
+  declare -Ar args_array=( [l]=logpath= [r]=failregex= [m]=max_retry= [p]=ports= )
+  local logpath
+  local failregex
+  local max_retry
+  local ports
+  # Manage arguments with getopts
+  ynh_handle_getopts_args "$@"
+  max_retry=${max_retry:-3}
+  ports=${ports:-http,https}
+
+  test -n "$logpath" || ynh_die "ynh_add_fail2ban_config expects a logfile path as first argument and received nothing."
+  test -n "$failregex" || ynh_die "ynh_add_fail2ban_config expects a failure regex as second argument and received nothing."
+
+  finalfail2banjailconf="/etc/fail2ban/jail.d/$app.conf"
+  finalfail2banfilterconf="/etc/fail2ban/filter.d/$app.conf"
+  ynh_backup_if_checksum_is_different "$finalfail2banjailconf" 1
+  ynh_backup_if_checksum_is_different "$finalfail2banfilterconf" 1
+
+  sudo tee $finalfail2banjailconf <<EOF
```

Comment:
As all script all called as `root` why does we continue to use sudo ?

Reply:
Probably because someone have forgotten to remove it.

Reply:
It could be good to clean this everywhere one day.